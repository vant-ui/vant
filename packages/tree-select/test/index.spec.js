import TreeSelect from '../';
import demoTest from '../../../test/demo-test';
import { mount } from '@vue/test-utils';

demoTest(TreeSelect);

test('empty list', () => {
  expect(mount(TreeSelect).html()).toMatchSnapshot();
});

test('select item', () => {
  const item = {
    text: 'city1',
    id: 1
  };
  const wrapper = mount(TreeSelect, {
    propsData: {
      items: [{
        text: 'group1',
        children: [item]
      }]
    }
  });

  wrapper.find('.van-tree-select__item').trigger('click');
  expect(wrapper.emitted('itemclick')[0][0]).toEqual(item);
});
