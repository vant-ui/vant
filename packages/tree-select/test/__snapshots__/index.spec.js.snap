// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty list 1`] = `
<div class="van-tree-select" style="height: 0px;">
  <div class="van-tree-select__nav"></div>
  <div class="van-tree-select__content" style="height: 0px;"></div>
</div>
`;

exports[`renders tree-select correctly 1`] = `
<div>
  <div>
    <div class="van-tree-select" style="height:300px;">
      <div class="van-tree-select__nav">
        <div class="van-ellipsis van-tree-select__nitem van-tree-select__nitem--active">
          所有城市
        </div>
        <div class="van-ellipsis van-tree-select__nitem">
          浙江
        </div>
        <div class="van-ellipsis van-tree-select__nitem">
          江苏
        </div>
      </div>
      <div class="van-tree-select__content" style="height:300px;">
        <div class="van-ellipsis van-tree-select__item van-tree-select__item--active">
          杭州
          <i class="van-icon van-icon-success van-tree-select__selected" style="color:undefined;">
            <!---->
          </i>
        </div>
        <div class="van-ellipsis van-tree-select__item">
          温州
          <!---->
        </div>
        <div class="van-ellipsis van-tree-select__item">
          宁波
          <!---->
        </div>
        <div class="van-ellipsis van-tree-select__item">
          义乌
          <!---->
        </div>
        <div class="van-ellipsis van-tree-select__item">
          无锡
          <!---->
        </div>
        <div class="van-ellipsis van-tree-select__item">
          常州
          <!---->
        </div>
        <div class="van-ellipsis van-tree-select__item">
          莆田
          <!---->
        </div>
        <div class="van-ellipsis van-tree-select__item">
          三明
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;
